### PR TITLE
feat: Frontend build generation workflow, switched to Terser from UglifyJs

### PR DESCRIPTION
**Pull request by sukuna28**
## What does it do?

This PR introduces a new build generation a workflow for frontend.

### CODE REVIEW

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10074867444). You can download and test it locally for verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10074876363). You can download and test it locally for verification.

File: .github/workflows/generate-frontend-build.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Continuous Integration for Frontend
+
+on:
+  push:
```
Comment (shahharsh176) [inline .github/workflows/generate-frontend-build.yml:4]: for this workflow, push event trigger is not required, right ?

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

File: .github/workflows/generate-frontend-build.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Building app
+        run: |
+          cd frontend
+          CI=false yarn build
+
+      - name: Upload build artifact
+        id: artifact-upload-step
+        uses: actions/upload-artifact@v4
+        with:
+          name: build
```
Comment (shahharsh176) [inline .github/workflows/generate-frontend-build.yml:45]: change name as per the pattern

File: .github/workflows/generate-frontend-build.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          path: backend/src/zango/assets/app_panel/js
+
+      - name: Get run ID
+        id: get_run_id
+        run: echo "run_id=${GITHUB_RUN_ID}" >> $GITHUB_OUTPUT
+
+      - name: Extract branch name
```
Comment (shahharsh176) [inline .github/workflows/generate-frontend-build.yml:52]: can remove this step

File: .github/workflows/generate-frontend-build.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          echo "Branch name is $branch_name"
+          echo "branch=$branch_name" >> $GITHUB_OUTPUT
+
+      - name: Add a comment to the PR
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.WORKFLOW_API_TOKEN }}
+          script: |
+            const prNumber = context.payload.pull_request.number;
+            const comment = "Thanks for the PR, we'll merge it after review.";
```
Comment (shahharsh176) [inline .github/workflows/generate-frontend-build.yml:66]: can remove

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

File: .github/workflows/generate-frontend-build.yml (hunk 1 above)
Comment (sukuna28) [reply .github/workflows/generate-frontend-build.yml:4]: yup, removed it

File: .github/workflows/generate-frontend-build.yml (hunk 2 above)
Comment (sukuna28) [reply .github/workflows/generate-frontend-build.yml:45]: done

File: .github/workflows/generate-frontend-build.yml (hunk 3 above)
Comment (sukuna28) [reply .github/workflows/generate-frontend-build.yml:52]: removed

File: .github/workflows/generate-frontend-build.yml (hunk 4 above)
Comment (sukuna28) [reply .github/workflows/generate-frontend-build.yml:66]: removed

File: .github/workflows/generate-frontend-build.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          github-token: ${{ secrets.WORKFLOW_API_TOKEN }}
+          script: |
+            const prNumber = context.payload.pull_request.number;
+            const artifactUrl = `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`;
+            await github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: prNumber,
+              body: `The build artifact is available in the artifact section [here](${artifactUrl}). You can download and test it locally for verification.`,
```
Comment (shahharsh176) [inline .github/workflows/generate-frontend-build.yml:57]: we can change comment to this

```text
The build artifact is available in the artifact section [here](${artifactUrl}). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.
```

File: .github/workflows/push-fe-build-to-main.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Building app
+        run: |
+          cd frontend
+          CI=false yarn build
+
+      - name: Setup Git for commit
+        run: |
+          git config user.name "github-actions"
+          git config user.email "actions@zango.com"
```
Comment (shahharsh176) [inline .github/workflows/push-fe-build-to-main.yml:40]: can we use `${{ secrets.USER_EMAIL }}` here? same as we have used in https://github.com/Healthlane-Technologies/Zango/blob/fe-build-test/.github/workflows/sync-docs-to-docs-hub-main.yml#L42C23-L42C48

**Review by shahharsh176 — CHANGES_REQUESTED** [review]

**Comment by shahharsh176** [comment]
@sukuna28 once we fix the issues and make the final changes, can you please rebase this branch with main and squash commits?

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10112603939). You can download and test it locally for verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10158232800). You can download and test it locally for verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10159646959). You can download and test it locally for verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10159764959). You can download and test it locally for verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10174251293). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Comment by kc-diabeat** [comment]
The build artifact is available in the artifact section [here](https://github.com/Healthlane-Technologies/Zango/actions/runs/10174485273). You can download it and place the build file under backend/src/zango/assets/app_panel/js for local testing and verification.

**Pull request closed without merging**